gadopt/utility.py

```diff
@@ -92,8 +92,17 @@ def update_timestep(self):
         return float(self.dt_const)
 
 
+def is_cartesian(mesh):
+    try:  # firedrake main
+        from firedrake import MeshSequenceGeometry  # noqa: F401
+
+        return mesh.unique().cartesian
+    except ImportError:  # firedrake release
+        return mesh.cartesian
```

Comment:
As per overall comment, could you please add the following to L25:
```
try:
    from firedrake import MeshSequenceGeometry
except ImportError:
    MeshSequenceGeometry = None
```

change `is_cartesian(mesh)` to this
```suggestion
def is_cartesian(mesh):
    if MeshSequenceGeometry is not None and isinstance(mesh, MeshSequenceGeometry):
        return mesh.unique().cartesian
    else:
        return mesh.cartesian
```
This fits more with our preferred coding style, and gives us a little extra sanity checking in case we ever want the `is_cartesian` function to work on other objects that have a `.cartesian` property.